lib/audit/auditor.go

```diff
@@ -190,9 +191,12 @@ func (s *ServerContext) Audit(w http.ResponseWriter, r *http.Request) {
 	logInfo.Printf("(%s) gcrPayload: %v", s.ID, gcrPayload)
 	logInfo.Printf("(%s) RemoteManifestFacility: %v", s.ID, s.RemoteManifestFacility)
 
-	// (3) Compare GCR state change with the intent of the promoter manifests.
+	// (3) Compare GCR state change with the intent of the promoter manifests
+	// (exact digest match on a tagged or tagless image).
 	for _, manifest := range manifests {
-		if manifest.Contains(*gcrPayload) {
+		m := gcrPayload.Match(manifest)
+		if m&(reg.FlagDigestMatched|reg.FlagTagMatched) != 0 &&
```

Comment:
Idea: I do wonder if the flags would be better expressed as a struct with bool members, particularly as some of them have true for match and some true for mismatch.

But ... what if match results `(bool, Info)`, where info was the bitset or struct with more info, and the bool was true if it's a match.